ratpack-core/src/main/java/ratpack/http/client/RequestSpec.java

```diff
@@ -117,6 +117,12 @@ default RequestSpec sslContext(Factory<SSLContext> factory) throws Exception {
 
   URI getUrl();
 
+  default RequestSpec connectTimeoutSeconds(int seconds) {
```

Comment:
I definitely agree on a default. I left it optional as I didn't want to change the default behaviour without getting a view from you guys.

Reading some of the Netty code, it appears the default connect timeout is 30000ms (https://github.com/netty/netty/blob/netty-4.1.0.Beta5/transport/src/main/java/io/netty/channel/DefaultChannelConfig.java) which is used by AbstractNioChannel connect (https://github.com/netty/netty/blob/netty-4.1.0.Beta5/transport/src/main/java/io/netty/channel/nio/AbstractNioChannel.java).

Will knock up a quick test to see what's actually going on but I think 30s sounds like a decent default (same as the read timeout in Ratpack from memory?)